### PR TITLE
HIVE-25111: Metastore Catalog Methods JDO Persistence

### PR DESCRIPTION
**Pull request by belugabehr**

### CODE REVIEW

File: standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/ObjectStore.java [hunk 1]
```diff
@@ -642,112 +642,140 @@ public void rollbackTransaction() {
...
+      if (tx.isActive()) {
+        tx.rollback();
       }
     }
   }
 
   @Override
   public void alterCatalog(String catName, Catalog cat)
-      throws MetaException, InvalidOperationException {
+      throws MetaException, InvalidOperationException, NoSuchObjectException {
```
Comment (belugabehr) [inline standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/ObjectStore.java:663]: This should also throw NoSuchObjectException if user is attempting to alter a non-existing catalog

File: standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/RawStore.java [hunk 2]
```diff
@@ -160,10 +160,11 @@
    * @param catName name of the catalog to alter.
    * @param cat new version of the catalog.
    * @throws MetaException something went wrong, usually in the database.
+   * @throws NoSuchObjectException no catalog of this name exists.
    * @throws InvalidOperationException attempt to change something about the catalog that is not
    * changeable, like the name.
    */
-  void alterCatalog(String catName, Catalog cat) throws MetaException, InvalidOperationException;
+  void alterCatalog(String catName, Catalog cat) throws MetaException, NoSuchObjectException, InvalidOperationException;
```
Comment (belugabehr) [inline standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/RawStore.java:167]: This should also throw NoSuchObjectException if user is attempting to alter a non-existing catalog

**Review by miklosgergely — APPROVED** [review]

**Comment by belugabehr** [comment]
@miklosgergely Nothing about this project can be easy. Oh well. Thanks for the review, but clearly this would be a bigger effort than I had imagined/hoped.

**Comment by belugabehr** [comment]
@miklosgergely  `ObjectStore` cannot decide what it wants to be.  Each method has its own transaction handling, but there are times, such as in `HMSHandler::create_catalog` that wraps several methods under one using its own transaction handling.  We need to decide which one it will be, it cannot be both.  Based on the current usage, it looks to me like the Transaction handling should be removed from the ObjectStore code and moved up to whatever is wrapping it (HMSHandler for example).

**Pull request closed without merging**